lisa/tools/tlb_stress.py

```diff
@@ -0,0 +1,1036 @@
+        """Install build dependencies for supported distributions only"""
+        from lisa.operating_system import CBLMariner, Debian
+
+        if isinstance(self.node.os, Debian):
+            self._log.debug("Detected Debian/Ubuntu distribution")
+            self._install_debian_deps()
+        elif isinstance(self.node.os, CBLMariner):
+            self._log.debug("Detected Azure Linux/CBL Mariner distribution")
+            self._install_azurelinux_deps()
+        else:
+            # Clear error message for unsupported distributions
+            distro_info = getattr(self.node.os, "name", "Unknown")
+            raise RuntimeError(
+                f"Unsupported distribution: {distro_info}. "
+                "This tool only supports Ubuntu (Debian) and Azure Linux "
+            )
+
+    def _install_debian_deps(self) -> None:
+        """Install dependencies for Debian/Ubuntu systems"""
+        packages = [
+            "build-essential",
+            "libc6-dev",
+            "linux-headers-generic",
+            "stress-ng",
+            "numactl",
+        ]
+        try:
+            packages.append("linux-tools-generic")
+        except Exception:
+            pass
+
+        self._log.debug("Installing Debian/Ubuntu build dependencies...")
+        self.node.os.install_packages(packages)  # type: ignore
+        self._log.debug("Successfully installed Debian/Ubuntu build dependencies")
+
+    def _install_azurelinux_deps(self) -> None:
+        """Install dependencies for Azure Linux/CBL Mariner systems"""
+        essential_packages = ["gcc", "make", "glibc-devel"]
+
+        # Try optional packages
+        self._try_install_package("glibc-headers", essential_packages)
+        self._try_install_kernel_headers(essential_packages)
+
+        # Install essential packages
+        self._log.debug("Installing Azure Linux/CBL Mariner build dependencies...")
+        self.node.os.install_packages(essential_packages)  # type: ignore
+
+        # Try optional packages
+        self._install_optional_packages()
+        self._log.debug(
+            "Successfully installed Azure Linux/CBL Mariner build dependencies"
+        )
+
+    def _try_install_package(self, package: str, package_list: List[str]) -> None:
+        """Try to install a package and add to list if successful"""
+        try:
+            self.node.os.install_packages([package])  # type: ignore
+            package_list.append(package)
+        except Exception:
+            self._log.debug(f"{package} not available as separate package")
+
+    def _try_install_kernel_headers(self, package_list: List[str]) -> None:
+        """Try to install kernel headers (kernel-headers or kernel-devel)"""
+        try:
+            self.node.os.install_packages(["kernel-headers"])  # type: ignore
+            package_list.append("kernel-headers")
+        except Exception:
+            try:
+                self.node.os.install_packages(["kernel-devel"])  # type: ignore
```

Comment:
It looks like it's handling different distros, so you can move those into the distro-specific install steps instead of placing them in `install_kernel_headers`, but not a general method.

Reply:
done

Reply:
Please check if it's addressed.